### PR TITLE
Respect avroname in sealed case object enums

### PR DESCRIPTION
**Pull request by alexjg**
I noticed that `@AvroName` is not respected in enum branches for `case object` style enums. I've added a few test cases and got them passing but I don't know if the solution I've come up with is the best one. Please let me know if there are changes I should make.

### CODE REVIEW

File: avro4s-macros/src/main/scala/com/sksamuel/avro4s/Decoder.scala [hunk 1]
```diff
@@ -289,8 +289,19 @@ object Decoder extends CoproductDecoders with TupleDecoders {
                                                                      toList: ToList[L, T]): Decoder[T] = new Decoder[T] {
     override def decode(value: Any, schema: Schema): T = {
       val name = value.toString
-      toList(objs()).find(_.toString == name).getOrElse(sys.error(s"Uknown type $name"))
+      val variants = toList(objs())
+      variants.find(v => {
+        println(getTypeName(v.getClass))
```
Comment (sksamuel) [inline avro4s-macros/src/main/scala/com/sksamuel/avro4s/Decoder.scala]: Should remove this.

**Comment by sksamuel** [comment]
I think it looks great bar leaving in the println.

**Comment by alexjg** [comment]
Whoops, removed!

**Comment by sksamuel** [comment]
Excellent

**Pull request merged**